.pre-commit-config.yaml

```diff
@@ -129,6 +129,14 @@ repos:
     rev: 37c2513b1b8275a475a160ed2f5b044910335d5f # No release tag yet including #6 fix
     hooks:
       - id: qml_formatter
+  - repo: https://github.com/gitleaks/gitleaks
+    rev: v8.24.0
+    hooks:
+      - id: gitleaks
```

Comment:
What would be the case for running Gitleaks in the case of a desktop application like Mixxx? As a rule of thumbs, I tend to reduce as much as possible software I would run from remote sources (which is what effectively pre-commit are), so I'd be interest to hear the pros for introducing this as mandatory hooks?

Reply:
Some time ago I wrote a helper script for the Launchpad-to-GitHub-Issues migration. I could have accidently committed the API credentials to the repo for example. If something like that is detected before committing, the creds are not actually exposed and leaked.

Reply:
Hmm fair enough. I'm still ensure of the net worth value, especially that we do have consensus to reduce resources footprint as part of #14297. Adding more hooks means more bandwidth and CPU time (both CI and on contributor machines) and I'm wondering if this justify these rares usecase. 
Please don't get me wrong, there is a lot of value in adding hooks as it means potentially one less push (and thus CI trigger), so I'm not again pre-commit hooks like the CMake one or the clazy one being introduced in #13642, but I feel like it is worth considering this. How about making it optional?

Reply:
Sure, just run `export SKIP=gitleaks` to disable it.

Reply:
this will disable it for other project on which `gitleaks` is relevant. Could we please adopt an opt-in policy instead, like our XML hook?